### PR TITLE
DAOS-10301 test: pass dmg command line with mmap()

### PR DESCRIPTION
**Pull request by bfaccini**
It seems possible that the buffer where dmg command line is being constructed, to be passed to system() call, can be in a page that had been registered for RDMA, which can cause a SEGV in fork()'ed child upon access for parsing.

So, to ensure allocating buffer in a brand new page, using mmap() to avoid this.

Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

### CODE REVIEW

**Comment by knard-intel** [comment]
From my side, do not understand how page allocated through malloc could be shared with memory mapped page ?
I am clearly not an expert in this area, but I was naively thinking that such situation could not happen: I was expecting that some kind of memory protection existing to prevent this.
Otherwise, your patch LGTM.

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Review by knard-intel — COMMENTED** [review]
LGTM thanks to @bfaccini explanation.

**Review by knard-intel — DISMISSED** [review]

**Comment by bfaccini** [comment]
> ftest LGTM

Hello @daltonbohning , I am not sure to understand the meaning of your comment/review ?! What do you mean by "ftest" and also why don't you review with a +1 if this PR looks good to you ?

**Comment by bfaccini** [comment]
> > ftest LGTM
> 
> Hello @daltonbohning , I am not sure to understand the meaning of your comment/review ?! What do you mean by "ftest" and also why don't you review with a +1 if this PR looks good to you ?

Hmm, more thinking about it, may be you were asking that the full CI pipeline/stages must have run before to declare this PR is ok ?!  So I have pushed a new commit without any commit-tag now to have a full CI testing session to occur for this PR !

File: src/tests/ftest/daos_test/suite.yaml [hunk 1]
```diff
@@ -166,6 +166,7 @@ daos_tests:
     test_daos_degraded_ec: X
     test_daos_dedup: U
   args:
+    test_daos_pool: -u6,7,9,13
```
Comment (kjacque) [inline src/tests/ftest/daos_test/suite.yaml:169]: Oops, this should be removed

**Review by kjacque — CHANGES_REQUESTED** [review]

File: src/tests/ftest/daos_test/suite.yaml (hunk 1 above)
Comment (bfaccini) [reply src/tests/ftest/daos_test/suite.yaml:169]: Oulah, right, I forgot this !

**Comment by bfaccini** [comment]
hmmm, the only 2 Existing failures : 

1)Test Hardware / Functional Hardware Large / 08-./daos_test/dfs.py – FTEST_launch.daos_test-dfs
2)Test Hardware / Functional Hardware Large / 10-./daos_racer/simple.py – FTEST_launch.daos_racer-simple

that have been triggered during last full CI session definitely look respectively related to both DAOS-11763/DAOS-11762 !!
Since associated patch/PR-10479 has has just landed to master, should I need to rebase and rerun CI again ???

**Review by knard-intel — APPROVED** [review]
OK from my side as it is just a merge with master.

**Pull request merged**